packages/starlight/utils/path.ts

```diff
@@ -50,3 +50,14 @@ export function ensureHtmlExtension(path: string) {
 	}
 	return ensureLeadingSlash(path);
 }
+
+export function splitQueryAndFragment(url: string) {
+	const [urlWithQuery, hash] = url.split('#');
+	const [urlBase, queryString] = (urlWithQuery || '').split('?');
+
+	return {
+		base: urlBase || '',
+		query: queryString ? `?${queryString}` : '',
+		fragment: hash ? `#${hash}` : '',
+	};
```

Comment:
But a second question: the usage in `favicon.ts` only needs the pathname right? We’re not using the search and hash?

In that case, a `stripSearchAndHash()` function might make more sense that just returns the pathname directly?

Reply:
@delucis - Agreed, in this case we only need `base`, keeping in mind it may be an absolute or relative url so I chose `base` vs. `pathname` since it could be more than just the "path" returned.  I included `query` and `fragment` in case it was ever needed in future since the parsing is required either way so seemed to make sense to return.  I chose `query` and `fragment` instead of using `search` and `hash` since this function, at least as written, separates a `url` in to its `parts` and `query` and `fragment` are the names given to the parts of a url (as opposed to the properties of URL itself which separates a url even further in to distinct properties).  All that is to say is why I chose what I did but happy to adjust however you think is best, just let me know what you think.

Reply:
Eliminated `splitQueryAndFragment` util function as it's no longer necessary given use of `URL` constructor.